### PR TITLE
Create yor_name tag

### PR DESCRIPTION
**Pull request by rotemavni**
### Created a new tag under code2cloud called yor_name

This tag tags each resource with their name.
The reason it's named `yor_name` is to avoid possible collisions with existing tags.

To support this I made. the following changes:
1. `GetResourceName` added to the IBlock interface and implemented differently for Terraform
2. Changed the `code2cloud` tag group to enable passing the block to the tagger.
3. Added tests.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/common/tagging/code2cloud/tag_group.go [hunk 1]
```diff
@@ -16,15 +18,33 @@ func (t *TagGroup) InitTagGroup(_ string, skippedTags []string, explicitlySpecif
...
+		&YorNameTag{},
 	}
 }
 
 func (t *TagGroup) CreateTagsForBlock(block structure.IBlock) error {
-	return t.UpdateBlockTags(block, struct{}{})
+	return t.UpdateBlockTags(block)
+}
+
+func (t *TagGroup) UpdateBlockTags(block structure.IBlock) error {
```
Comment (nimrodkor) [inline src/common/tagging/code2cloud/tag_group.go:35]: @rotemavni Why is this needed? What were we missing?

**Review by nimrodkor — COMMENTED** [review]
Nice!

File: src/common/tagging/code2cloud/tag_group.go (hunk 1 above)
Comment (rotemavni) [reply src/common/tagging/code2cloud/tag_group.go:35]: @nimrodkor This method is implemented in the base block, but doesn't pass the `block` into `tag.CalculateValue(block)`. We need to pass it on so we can determine the block name for the tag.

**Review by tronxd — APPROVED** [review]

**Review by nimrodkor — APPROVED** [review]
